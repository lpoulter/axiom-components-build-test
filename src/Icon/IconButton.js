import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Button from '../Button/Button';
import ButtonIcon from '../Button/ButtonIcon';

const sizeMap = {
  'small': '1rem',
  'medium': '2rem',
  'large': '3rem',
  'huge': '5rem',
};

export default class IconButton extends Component {
  static propTypes = {
    /** Style of the Button, which affects its coloring and sizing */
    buttonStyle: PropTypes.oneOf(['primary', 'secondary', 'tertiary', 'quaternary']),
    /** Color of the icon */
    iconColor: PropTypes.oneOf([
      'body',
      'day',
      'night',
      'error',
      'success',
      'warning',
      'disabled',
      'subtle',
      'twitter',
      'facebook',
      'instagram',
      'forbidden-planet',
      'tiny-clanger',
      'critical-mass',
      'fantastic-voyage',
      'paradise-lost',
      'serene-sea',
      'event-horizon',
      'electric-dreams',
      'outer-limits',
      'giant-leap',
      'moon-lagoon',
      'space-invader',
      'extraterrestrial',
      'terra-form',
      'primeval-soup',
      'future-shock',
      'sun-maker',
      'new-horizon',
      'blast-off',
      'crash-course',
      'solar-rust',
      'ground-control',
      'space-oddity',
      'rocky-planet',
      'deep-thought',
      'luna-dust',
      'sentiment-negative',
      'sentiment-positive',
      'sentiment-neutral',
      'ui-accent',
    ]),
    /** Name of the icon that will be displayed */
    name: PropTypes.string.isRequired,
    /** Size of the button and icon */
    size: PropTypes.oneOf(['small', 'medium', 'large', 'huge']),
  };

  static defaultProps = {
    size: 'medium',
    buttonStyle: 'secondary',
  }

  render() {
    const { name, size, buttonStyle, iconColor, ...rest } = this.props;

    return (
      <Button { ...rest } shape="circle" size={ size } style={ buttonStyle }>
        <ButtonIcon color={ iconColor } name={ name } size={ sizeMap[size] } />
      </Button>
    );
  }
}
