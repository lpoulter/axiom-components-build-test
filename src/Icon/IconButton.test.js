import React from 'react';
import renderer from 'react-test-renderer';
import IconButton from './IconButton';

const getComponent = (props = {}) =>
  renderer.create(
    <IconButton name="twitter" { ...props } />
  );

describe('IconButton', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with size', () => {
    ['small', 'medium', 'large', 'huge'].forEach((size) => {
      it(size, () => {
        const component = getComponent({ size });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
