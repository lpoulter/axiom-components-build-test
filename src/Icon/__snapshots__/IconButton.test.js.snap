// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders with defaultProps 1`] = `
<button
  className="ax-button ax-button--accent ax-button--secondary ax-button--circle-medium"
>
  <span
    className="ax-button__icon ax-button__icon--start ax-button__icon--end"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "2rem",
          "width": "2rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </span>
</button>
`;

exports[`IconButton renders with size huge 1`] = `
<button
  className="ax-button ax-button--accent ax-button--secondary ax-button--circle-huge"
>
  <span
    className="ax-button__icon ax-button__icon--start ax-button__icon--end"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "5rem",
          "width": "5rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </span>
</button>
`;

exports[`IconButton renders with size large 1`] = `
<button
  className="ax-button ax-button--accent ax-button--secondary ax-button--circle-large"
>
  <span
    className="ax-button__icon ax-button__icon--start ax-button__icon--end"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "3rem",
          "width": "3rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </span>
</button>
`;

exports[`IconButton renders with size medium 1`] = `
<button
  className="ax-button ax-button--accent ax-button--secondary ax-button--circle-medium"
>
  <span
    className="ax-button__icon ax-button__icon--start ax-button__icon--end"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "2rem",
          "width": "2rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </span>
</button>
`;

exports[`IconButton renders with size small 1`] = `
<button
  className="ax-button ax-button--accent ax-button--secondary ax-button--circle-small"
>
  <span
    className="ax-button__icon ax-button__icon--start ax-button__icon--end"
  >
    <svg
      className="ax-icon ax-icon--twitter"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </span>
</button>
`;
